packages/aws_signature_v4/test/c_test_suite/test_data_loader.dart

```diff
@@ -0,0 +1,193 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import 'context.dart';
+import 'request_parser.dart';
+import 'test_data.dart';
+
+/// Handles parsing for one file.
+typedef TestFileParser = Future<void> Function(SignerTestBuilder, String);
+
+/// File parsers for the current test suite directory organization.
+const Map<String, TestFileParser> _fileParsers = {
+  'context.json': _parseContext,
+  'request.txt': _parseRequest,
+  'query-canonical-request.txt': _parseQueryCanonicalRequest,
+  'query-string-to-sign.txt': _parseQueryStringToSign,
+  'query-signature.txt': _parseQuerySignature,
+  'query-signed-request.txt': _parseQuerySignedRequest,
+  'header-canonical-request.txt': _parseHeaderCanonicalRequest,
+  'header-string-to-sign.txt': _parseHeaderStringToSign,
+  'header-signature.txt': _parseHeaderSignature,
+  'header-signed-request.txt': _parseHeaderSignedRequest,
+};
+
+/// Parses raw HTTP requests.
+final _requestParser = SignerRequestParser();
+
+/// Loads all test cases in the C signer test suite by walking the file tree
+/// and creating a [SignerTest] for each folder encountered where all requisite
+/// files are available.
+Future<List<SignerTest>> loadAllTests() async {
+  await _requestParser.init();
+
+  final testSuitePath = path.joinAll([
+    Directory.current.path,
+    'external',
+    'aws-c-auth',
+    'tests',
+    'aws-signing-test-suite',
+    'v4',
+  ]);
+  final testCases = <SignerTest>[];
+
+  for (var entity in Directory(testSuitePath).listSync()) {
+    final stat = entity.statSync();
+    if (stat.type != FileSystemEntityType.directory) {
+      throw StateError('Unexpected top level file found: ${entity.path}');
+    }
+    final testCaseDir = Directory(entity.path);
+    final testCaseName = path.basename(entity.path);
+    final testCaseBuilder = SignerTestBuilder(testCaseName);
+
+    // List all the files in the test directory and sort them alphabetically.
+    // This is required to ensure that `context.json` is the first file parsed,
+    // since it's data is used to parse other files accordingly, namely `request.txt`.
+    final testFiles =
+        testCaseDir.listSync().map((ent) => File(ent.path)).toList()
+          ..sort(
+            (a, b) => path.basename(a.path).compareTo(
+                  path.basename(b.path),
+                ),
+          );
+
+    // Some folders do not include all the necessary files. Not sure why - maybe
+    // they are borrowing the rest of the files from other folders, but it's not
+    // clear. This represents only 2 tests, as of writing.
```

Comment:
Isn't it important that we figure out what those 2 test suites are for? Also, can/should we assert that this is the case that only 2 test suites are failing to run due to this limitation? If for some reason in the future more requisite files went missing, we wouldn't want to just happily skip the test without an investigation being prompted I think?

Reply:
I opened an issue with them. Based on their feedback, I'll update as appropriate: https://github.com/awslabs/aws-c-auth/issues/160

Reply:
Should we still assert that 8 tests *are* running?

Reply:
Probably not a huge deal. We're locked to a commit of aws-c-auth since it's a submodule. If anything changes, we'd know since we'd have to manually update the ref with a PR.

Reply:
Yeah I'm just thinking about someone who's never seen any of this coming in and going oh I can just update the submodule and unknowingly skipped more tests. Not a huge deal - just keeping an eye out for maintainability